packages/token-bridge-sdk/src/hooks/arbTokenBridge.types.ts

```diff
@@ -48,24 +48,75 @@ export enum AssetType {
 
 export type TransactionLifecycle<Tx, TxReceipt> = Partial<{
   onTxSubmit: (tx: Tx) => void
-  onTxConfirm: (txReceipt: TxReceipt) => void
+  onTxConfirm: (tx: Tx, txReceipt: TxReceipt) => void
 }>
 
-export type L1EthDepositTransactionLifecycle = TransactionLifecycle<
-  L1EthDepositTransaction,
-  L1EthDepositTransactionReceipt
->
+export type TokenTransactionLifecycle<Tx, TxReceipt> = Pick<
+  TransactionLifecycle<Tx, TxReceipt>,
+  'onTxConfirm'
+> &
+  Partial<{
+    onTxSubmit: (tx: Tx, symbol: string) => void
+  }>
 
-export type L1ContractCallTransactionLifecycle = TransactionLifecycle<
-  L1ContractCallTransaction,
-  L1ContractCallTransactionReceipt
->
+export type L1EthDepositTransactionLifecycle = Pick<
+  TransactionLifecycle<L1EthDepositTransaction, L1EthDepositTransactionReceipt>,
+  'onTxSubmit'
+> & {
+  onTxConfirm: (
+    tx: L1EthDepositTransaction,
+    txReceipt: L1EthDepositTransactionReceipt,
+    ethDepositMessage: EthDepositMessage
+  ) => void
+}
+
+export type TokenL1ContractCallTransactionLifecycle = Pick<
+  TokenTransactionLifecycle<
+    L1ContractCallTransaction,
+    L1ContractCallTransactionReceipt
+  >,
+  'onTxSubmit'
+> & {
+  onTxConfirm: (
+    tx: L1ContractCallTransaction,
+    txReceipt: L1ContractCallTransactionReceipt,
+    l1Tol2Message: IL1ToL2MessageReader
+  ) => void
+}
```

Comment:
Similar ETH deposit above, let's do `onL1TxSubmit`, `onL1TxSuccess`, `onL1TxFailure` and `onL1ToL2Message`

Reply:
Also within the context of Bridge UI, I think it makes sense to name the type `TokenDepositTransactionLifecycle `